contracts/utils/CREATE2Factory.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title CREATE2Factory
+ *
+ * @dev This contract factory focus on the use of CREATE2 EVM opcode for
+ * contracts deployment. It also provides a function to precompute the address
+ * where the smart contracts with the specified salt and bytecode would be
+ * deployed.
+ */
+contract CREATE2Factory {
+
+    // Event triggered when a contract is deployed
+    event Deployed(address addr, bytes32 salt);
+
+    /**
+     * @dev Deploy contract with CREATE2
+     * @param salt The salt used to the contract address computation
+     * @param code The bytecode of of the contract to be deployed
+     */
+    function deploy(bytes32 salt, bytes memory code) public {
+        address addr = _deploy(salt, code);
+        emit Deployed(addr, salt);
+    }
+
+
+    /**
+     * @dev Function to compute the address of a contract created with CREATE2.
+     * @param deployer The address of the smart contract that will deploy the
+     * contract
+     * @param salt The salt used to the contract address computation
+     * @param code The bytecode of the contract to be deployed
+     * @return the computed address of the smart contract.
+     */
+    function computeAddress(
+        address deployer, bytes32 salt, bytes memory code
+    ) public view returns (address) {
+        bytes32 codeHash = keccak256(code);
+        bytes32 _data = keccak256(
+            abi.encodePacked(bytes1(0xff), deployer, salt, codeHash)
```

Comment:
I'm confused, shouldn't `deployer` be `address(this)`?

Reply:
@nventuro Agreed it should be `address(this)`

Reply:
yes, my bad. changing it